### PR TITLE
Full lab without solution

### PR DESCRIPTION
**Pull request by changemewtf**

### CODE REVIEW

**Comment by loganhasson** [comment]
Let's make sure to give students the `models` directory

File: spec/controllers/songs_controller_spec.rb [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+      released: true
+    }
+  end
+
+  describe "basic listing, viewing and editing" do
+    let(:song) { Song.create!(valid_attributes) }
+
+    it "lists all songs" do
+      get :index
+      expect(assigns(:songs)).to eq([song])
```
Comment (loganhasson) [inline spec/controllers/songs_controller_spec.rb:28]: Let's use alternatives to `assigns`.

File: spec/requests/songs_spec.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+require 'rails_helper'
+
+RSpec.describe "Songs", type: :request do
```
Comment (loganhasson) [inline spec/requests/songs_spec.rb]: this file/directory should get ripped out

File: Gemfile [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+gem 'rails', '4.2.5'
+gem 'sqlite3'
+gem 'rspec-rails'
+gem 'sass-rails', '~> 5.0'
+gem 'uglifier', '>= 1.3.0'
+gem 'jbuilder', '~> 2.0'
+gem 'sdoc', '~> 0.4.0', group: :doc
+
+group :development, :test do
+  gem 'byebug'
```
Comment (loganhasson) [inline Gemfile]: let's include 'pry'

File: README.md [hunk 4]
```diff
@@ -1,24 +1,37 @@
-# Crud With Validations Lab
+# CRUD With Validations Lab
 
-## Objectives
+Your goal in this lab is to create a thorough CRUD interface for one model, the
```
Comment (loganhasson) [inline README.md:3]: I'd like to see more of a story here. Let's think of a way to describe this as a feature the student is building, not just a list of tasks. Maybe they're building a basic system to catalog their music or something for a record exec to keep track of their artists' releases. Something to jazz it up a bit.

**Comment by loganhasson** [comment]
As I was working through this lab, I liked the progression of the tests. A couple of notes:

1) let's stick to just controller/model/feature specs.
2) instead of rails-dom-testing, please use capybara for testing elements on the page (i found the initial error messages from the view tests really hard to parse)

File: README.md [hunk 5]
```diff
@@ -1,24 +1,37 @@
...
+  - Must be less than or equal to the current year
+- `artist_name`, a `string`
+  - Must not be blank
+- `genre`, a `string`
 
-the models should have some validations (unique artist name, song title presence, etc)
+# Requirements
 
-forms should require some customization in terms of validation messages or errors and just something to make it a little custom.
+Use the `resource` generator, **not** the `scaffold` generator.
```
Comment (loganhasson) [inline README.md:26]: might be worth giving a really, really basic example just to refresh their memory of what the resource generator looks like

File: spec/controllers/songs_controller_spec.rb (hunk 1 above)
Comment (changemewtf) [inline spec/controllers/songs_controller_spec.rb:28]: What do you recommend here?

If we test the HTML, then we're duplicating `spec/views/songs/index.html.erb_spec.rb`.

If we test the ActiveRecord functions, then we're not actually testing the controller at all.

File: README.md (hunk 5 above)
Comment (changemewtf) [inline README.md:26]: This is the rails error:

```
     Minitest::Assertion:
       Expected at least 1 element matching "form[action="/songs/2"]", found 0..
       Expected 0 to be >= 1.
```

If I replace the `assert_select` with Capybara's `have_selector` like so:

``` ruby
 expect(rendered).to have_selector("form[action='#{song_path(song)}']")
```

We get the error:

```
  1) songs/edit renders the edit song form
     Failure/Error: expect(rendered).to have_selector("form[action='a#{song_path(song)}']")
       expected to find css "form[action='a/songs/1']" but there were no matches
```

I don't personally find that any clearer, so I wanted to check with you before changing everything, because it will probably take a while.

**Comment by jmburges** [comment]
looks good to me!
